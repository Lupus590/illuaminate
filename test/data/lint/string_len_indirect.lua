-- config: (lint (only (stdlib:string-len)))

local x = string
local y = string.len
local z = x.len

local _ = x.len "foo"
local _ = y "foo"
local _ = z "foo"
