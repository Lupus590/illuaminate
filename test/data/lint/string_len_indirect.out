string_len_indirect.lua:[7:11-7:22]: Prefer `#` over string.len [stdlib:string-len]
   │
 7 │ local _ = x.len "foo"
   │           ^^^^^^^^^^^
string_len_indirect.lua:[8:11-8:18]: Prefer `#` over string.len [stdlib:string-len]
   │
 8 │ local _ = y "foo"
   │           ^^^^^^^
string_len_indirect.lua:[9:11-9:18]: Prefer `#` over string.len [stdlib:string-len]
   │
 9 │ local _ = z "foo"
   │           ^^^^^^^
No errors and 3 warnings
@@ -4,7 +4,7 @@
 local y = string.len
 local z = x.len
 
-local _ = x.len "foo"
-local _ = y "foo"
-local _ = z "foo"
+local _ = #"foo"
+local _ = #"foo"
+local _ = #"foo"
 
